File: lab09_Mosque_Management/untitled/src/WaqtTimeTablesWidget.java
import java.util.HashMap;
import java.util.Map;
import java.time.LocalTime;

class WaqtTimeTablesWidget extends Component {
    private Map<String, LocalTime> waqtTimes = new HashMap<>();

    public void setWaqtTimes(Map<String, LocalTime> waqtTimes) {
        this.waqtTimes = waqtTimes;
        mediator.notify(this, "waqtUpdated");
    }

    public Map<String, LocalTime> getWaqtTimes() {
        return waqtTimes;
    }

    @Override
    public void update() {
        System.out.println("Waqt Time Tables updated: " + waqtTimes);
    }
}

File: Lab9_Mosque_Management_System/Lab9_Mosque_Management_System/Widgets/WaqtTimeTablesWidget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab9_Mosque_Management_System
{
    public class WaqtTimeTablesWidget : IWidget
    {
        private IMediator _mediator;

        public void SetMediator(IMediator mediator)
        {
            _mediator = mediator;
        }

        public void Update()
        {
            Console.WriteLine("Waqt Time Tables updated based on Location and TimeZone.");
            _mediator.Notify(this, "WaqtTimeUpdated");
        }
    }
}
